diffpy/snmf/datapoint.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+import numdifftools
+
+
+class ComponentSignal:
+    """
+    Attributes
+    ----------
+    iq: 1d array of floats
+      The intensity/g(r) values of the component
+    grid: 1d array of floats
+      The vector containing the grid points of the component
+    weights: 1d array of floats
+      The vector containing the weight of the component signal for each signal
+    stretching_factors: 1d array of floats
+      The vector containing the stretching factor for the component signal for each signal
+    id: int
+      The component number.
+    """
+    def __init__(self, grid, iq, weights, stretching_factors, id_number):
+        self.iq = np.asarray(iq)
```

Comment:
it may be a bit confusing to a code reader to have the attributes defined in a different order than they are in the signature.

Reply:
Ok. I will change the order